File: frontend-admin/src/types/document.ts
export interface Document {
    id: string;
    filename: string;
    content_type: string;
    file_size: number;
    created_at: string;
    status: 'pending' | 'processing' | 'completed' | 'failed';
    error_message?: string;
}

export interface DocumentUploadResponse {
    id: string;
    filename: string;
    status: string;
}

File: backend/app/schemas/document.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from uuid import UUID
from app.models.document import DocumentStatus

class DocumentResponse(BaseModel):
    id: UUID
    filename: str
    content_type: Optional[str] = None
    file_size: Optional[int] = None
    status: DocumentStatus
    error_message: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class DocumentUploadResponse(BaseModel):
    document_id: UUID
    filename: str
    status: DocumentStatus
    message: str

class DocumentListResponse(BaseModel):
    items: List[DocumentResponse]
    total: int
